pdm/cli/commands/run.py

```diff
@@ -151,11 +152,16 @@ def _run_process(
                 process_env["NO_SITE_PACKAGES"] = "1"
 
         cwd = project.root if chdir else None
-        process = subprocess.Popen(expanded_args, cwd=cwd, env=process_env, shell=shell)
-        try:
-            process.wait()
-        except KeyboardInterrupt:
-            pass
+
+        def send_int(exitnum, handler):
+            process.send_signal(signal.SIGINT)
+
+        s = signal.signal(signal.SIGINT, send_int)
```

Comment:
A lambda is enough: `lambda signum, frame: process.send_signal(signum)`

Reply:
Yeah, I didn't know if that was important enough to annotate it.

Reply:
Yeah, if you use a normal function you need to annotate it, the signature is `(signum: int, frame: types.FrameType|None) -> None`